homeassistant/components/telegram_bot/polling.py

```diff
@@ -91,30 +96,43 @@ def get_updates(self, offset):
                 )
             if resp.status == 200:
                 _json = yield from resp.json()
+                return _json
             else:
-                _LOGGER.error("Error %s on %s", resp.status, self.update_url)
-
-        except ValueError:
-            _LOGGER.error("Error parsing Json message")
-        except (asyncio.TimeoutError, ClientError):
-            _LOGGER.error("Client connection error")
+                raise WrongHttpStatus('wrong status %s', resp.status)
         finally:
             if resp is not None:
                 yield from resp.release()
 
-        return _json
-
     @asyncio.coroutine
     def handle(self):
         """Receiving and processing incoming messages."""
-        _updates = yield from self.get_updates(self.update_id)
-        _updates = _updates.get('result')
-        if _updates is None:
-            _LOGGER.error("Incorrect result received.")
+        try:
+            _updates = yield from self.get_updates(self.update_id)
+        except WrongHttpStatus as err:
+            # WrongHttpStatus: Non-200 status code.
+            # Occurs at times (mainly 502) and recovers
+            # automatically. Pause for a while before retrying.
+            _LOGGER.error(err)
+            yield from asyncio.sleep(RETRY_SLEEP)
```

Comment:
Solve that a level higher

Reply:
@pvizeli What exactly do you mean ? Should I move all exceptions somewhere else or is it only about the WrongHttpStatus ?

Reply:
Extract the logic into `check_incoming`